### PR TITLE
Change project-wide `Person` to `Property` and add `Seller`, `Price`

### LINKED ISSUE #25: Rename model from Person to Property
Rename model from `Person` to `Property` and refactor codebase to make the code compile

### PR DESCRIPTION
**Pull request by trxe**
closes #25 
@waynetee pls help review thanks

### CODE REVIEW

File: src/main/java/seedu/address/logic/Logic.java [hunk 1]
```diff
@@ -30,8 +30,8 @@
      */
     ReadOnlyAddressBook getAddressBook();
 
-    /** Returns an unmodifiable view of the filtered list of persons */
-    ObservableList<Person> getFilteredPersonList();
+    /** Returns an unmodifiable view of the filtered list of properties */
+    ObservableList<Property> getFilteredPersonList();
```
Comment (guangxuan) [inline src/main/java/seedu/address/logic/Logic.java:34]:
```suggestion
    ObservableList<Property> getFilteredPropertyList();
```

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 2]
```diff
@@ -5,56 +5,64 @@
...
-            + PREFIX_TAG + "owesMoney";
+            + PREFIX_TAG + "Condo"
+            + PREFIX_TAG + "4 rm"
+            + PREFIX_TAG + "621 sqft"
+            + PREFIX_TAG + "EW23 Clementi";
 
-    public static final String MESSAGE_SUCCESS = "New person added: %1$s";
-    public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book";
+    public static final String MESSAGE_SUCCESS = "New property added: %1$s";
+    public static final String MESSAGE_DUPLICATE_PERSON = "This property already exists in the address book";
```
Comment (guangxuan) [inline src/main/java/seedu/address/logic/commands/AddCommand.java:45]:
```suggestion
    public static final String MESSAGE_DUPLICATE_PROPERTY = "This property already exists in the address book";
```

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 3]
```diff
@@ -19,87 +21,95 @@
...
 
-    public static final String MESSAGE_EDIT_PERSON_SUCCESS = "Edited Person: %1$s";
+    public static final String MESSAGE_EDIT_PROPERTY_SUCCESS = "Edited Property: %1$s";
     public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
-    public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book.";
+    public static final String MESSAGE_DUPLICATE_PROPERTY = "This property already exists in the address book.";
 
     private final Index index;
-    private final EditPersonDescriptor editPersonDescriptor;
+    private final EditPropertyDescriptor editPersonDescriptor;
```
Comment (guangxuan) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:60]:
```suggestion
    private final EditPropertyDescriptor editPropertyDescriptor;
```

File: src/main/java/seedu/address/logic/parser/EditCommandParser.java [hunk 4]
```diff
@@ -42,7 +42,7 @@ public EditCommand parse(String args) throws ParseException {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE), pe);
         }
 
-        EditPersonDescriptor editPersonDescriptor = new EditPersonDescriptor();
+        EditCommand.EditPropertyDescriptor editPersonDescriptor = new EditPropertyDescriptor();
```
Comment (guangxuan) [inline src/main/java/seedu/address/logic/parser/EditCommandParser.java:45]:
```suggestion
        EditCommand.EditPropertyDescriptor editPropertyDescriptor = new EditPropertyDescriptor();
```

File: src/main/java/seedu/address/model/AddressBook.java [hunk 5]
```diff
@@ -5,16 +5,16 @@
...
 
 import javafx.collections.ObservableList;
-import seedu.address.model.person.Person;
-import seedu.address.model.person.UniquePersonList;
+import seedu.address.model.property.Property;
+import seedu.address.model.property.UniquePropertyList;
 
 /**
  * Wraps all data at the address-book level
  * Duplicates are not allowed (by .isSamePerson comparison)
```
Comment (guangxuan) [inline src/main/java/seedu/address/model/AddressBook.java:13]:
```suggestion
 * Duplicates are not allowed (by .isSameProperty comparison)
```

File: src/main/java/seedu/address/storage/JsonAdaptedProperty.java [hunk 6]
```diff
@@ -60,11 +69,11 @@ public JsonAdaptedPerson(Person source) {
...
     /**
-     * Converts this Jackson-friendly adapted person object into the model's {@code Person} object.
+     * Converts this Jackson-friendly adapted property object into the model's {@code Property} object.
      *
-     * @throws IllegalValueException if there were any data constraints violated in the adapted person.
+     * @throws IllegalValueException if there were any data constraints violated in the adapted property.
      */
-    public Person toModelType() throws IllegalValueException {
+    public Property toModelType() throws IllegalValueException {
         final List<Tag> personTags = new ArrayList<>();
```
Comment (guangxuan) [inline src/main/java/seedu/address/storage/JsonAdaptedProperty.java:77]:
```suggestion
        final List<Tag> propertyTags = new ArrayList<>();
```

File: src/main/java/seedu/address/storage/JsonSerializableAddressBook.java [hunk 7]
```diff
@@ -11,24 +11,24 @@
...
+import seedu.address.model.property.Property;
 
 /**
  * An Immutable AddressBook that is serializable to JSON format.
  */
 @JsonRootName(value = "addressbook")
 class JsonSerializableAddressBook {
 
-    public static final String MESSAGE_DUPLICATE_PERSON = "Persons list contains duplicate person(s).";
+    public static final String MESSAGE_DUPLICATE_PERSON = "Property list contains duplicate property(s).";
```
Comment (guangxuan) [inline src/main/java/seedu/address/storage/JsonSerializableAddressBook.java:22]:
```suggestion
    public static final String MESSAGE_DUPLICATE_PROPERTY = "Property list contains duplicate property(s).";
```

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 8]
```diff
@@ -110,8 +110,8 @@ private void setAccelerator(MenuItem menuItem, KeyCombination keyCombination) {
      * Fills up all the placeholders of this window.
      */
     void fillInnerParts() {
-        personListPanel = new PersonListPanel(logic.getFilteredPersonList());
-        personListPanelPlaceholder.getChildren().add(personListPanel.getRoot());
+        propertyListPanel = new PropertyListPanel(logic.getFilteredPersonList());
+        personListPanelPlaceholder.getChildren().add(propertyListPanel.getRoot());
```
Comment (guangxuan) [inline src/main/java/seedu/address/ui/MainWindow.java:114]:
```suggestion
        propertyListPanelPlaceholder.getChildren().add(propertyListPanel.getRoot());
```

File: src/main/java/seedu/address/ui/PropertyCard.java [hunk 9]
```diff
@@ -42,17 +42,17 @@
     private FlowPane tags;
 
     /**
-     * Creates a {@code PersonCode} with the given {@code Person} and index to display.
+     * Creates a {@code PersonCode} with the given {@code Property} and index to display.
```
Comment (guangxuan) [inline src/main/java/seedu/address/ui/PropertyCard.java:45]:
```suggestion
     * Creates a {@code PropertyCode} with the given {@code Property} and index to display.
```

**Review by guangxuan — CHANGES_REQUESTED** [review]
Generally good work in refactoring. A few unchanged Person keywords within variable/function names. 
Not sure whether Seller should reuse Name class or not. @waynetee what do you think?

**Comment by waynetee** [comment]
> Generally good work in refactoring. A few unchanged Person keywords within variable/function names.
> Not sure whether Seller should reuse Name class or not. @waynetee what do you think?

I think it is ok for Seller not to reuse the name for now, we can do more refactoring later on. I'll prefer to not delay the merging of this pr.

@trxe when you are done with the changes you can request a review from me and/or guangxuan using the reviewers panel on the top right of the webpage.

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 10]
```diff
@@ -163,8 +163,8 @@ private void handleExit() {
         primaryStage.hide();
     }
 
-    public PersonListPanel getPersonListPanel() {
-        return personListPanel;
+    public PropertyListPanel getPersonListPanel() {
```
Comment (guangxuan) [inline src/main/java/seedu/address/ui/MainWindow.java:166]:
```suggestion
    public PropertyListPanel getPropertyListPanel() {
```

**Review by guangxuan — APPROVED** [review]
LGTM. Only left with 1 occurrence of person within the main body of code. The rest (in the tests) can be done on another PR.

File: build.gradle [hunk 11]
```diff
@@ -41,6 +41,7 @@ task coverage(type: JacocoReport) {
 }
 
 dependencies {
+    implementation 'junit:junit:4.13.1'
     String jUnitVersion = '5.4.0'
```
Comment (waynetee) [inline build.gradle:45]: this seems weird, can we stick to junit 5?
Might need to change all the `@Ignore` to `@Disabled`

File: src/main/java/seedu/address/model/property/Property.java [hunk 12]
```diff
@@ -60,57 +64,70 @@ public Address getAddress() {
...
+                && otherProperty.getEmail().equals(getEmail())
+                && otherProperty.getAddress().equals(getAddress())
+                && otherProperty.getTags().equals(getTags());
     }
 
     @Override
     public int hashCode() {
         // use this method for custom fields hashing instead of implementing your own
-        return Objects.hash(name, phone, email, address, tags);
+        return Objects.hash(name, phone, email, address, tags, seller);
```
Comment (waynetee) [inline src/main/java/seedu/address/model/property/Property.java:114]: what about the price?

**Review by waynetee — CHANGES_REQUESTED** [review]
image
You can use ctrl-shif-f on intelij to find all remaining occurrences of "person"

File: src/main/java/seedu/address/model/property/Property.java (hunk 12 above)
Comment (trxe) [reply src/main/java/seedu/address/model/property/Property.java:114]: IMO Price shouldn't be tied to identity of the Property, i.e. two properties with everything identical except price should still be considered identical.

File: build.gradle (hunk 11 above)
Comment (trxe) [reply build.gradle:45]: This was what it came as, maybe we can discuss a bit more first?

File: src/main/java/seedu/address/model/property/Property.java (hunk 12 above)
Comment (trxe) [reply src/main/java/seedu/address/model/property/Property.java]: I've made the change to include price.

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 13]
```diff
@@ -19,87 +21,95 @@
...
+            editPersonDescriptor) {
+        assert propertyToEdit != null;
+
+        Name updatedName = editPersonDescriptor.getName().orElse(propertyToEdit.getName());
+        Seller updatedSeller = editPersonDescriptor.getSeller().orElse(propertyToEdit.getSeller());
+        Price updatedPrice = editPersonDescriptor.getPrice().orElse(propertyToEdit.getPrice());
+        Phone updatedPhone = editPersonDescriptor.getPhone().orElse(propertyToEdit.getPhone());
+        Email updatedEmail = editPersonDescriptor.getEmail().orElse(propertyToEdit.getEmail());
+        Address updatedAddress = editPersonDescriptor.getAddress().orElse(propertyToEdit.getAddress());
+        Set<Tag> updatedTags = editPersonDescriptor.getTags().orElse(propertyToEdit.getTags());
```
Comment (waynetee) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:109]: Can you remove remaining occurrences of person here and where mentioned by Guangxuan (excluding test code)?
Use the IDE to help you

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 14]
```diff
@@ -209,14 +239,15 @@ public boolean equals(Object other) {
...
 
             // state check
-            EditPersonDescriptor e = (EditPersonDescriptor) other;
+            EditPropertyDescriptor e = (EditPropertyDescriptor) other;
 
             return getName().equals(e.getName())
+                    && getSeller().equals(e.getSeller())
                     && getPhone().equals(e.getPhone())
                     && getEmail().equals(e.getEmail())
                     && getAddress().equals(e.getAddress())
```
Comment (waynetee) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:253]: same issue with missing price check

**Review by waynetee — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/model/property/UniquePropertyList.java [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.property.exceptions.DuplicatePropertyException;
+import seedu.address.model.property.exceptions.PropertyNotFoundException;
+
+/**
+ * A list of properties that enforces uniqueness between its elements and does not allow nulls.
+ * A property is considered unique by comparing using {@code Property#isSamePerson(Property)}.
+ * As such, adding and updating of properties uses Property#isSamePerson(Property) for equality
```
Comment (waynetee) [inline src/main/java/seedu/address/model/property/UniquePropertyList.java:17]: Replace person with property here

**Review by waynetee — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**